components/core/tests/test-GlobalMetadataDBConfig.cpp

```diff
@@ -0,0 +1,132 @@
+            "--db-port",
+            "8888",
+            "--db-name",
+            "test-db",
+            "--db-table-prefix",
+            "test_prefix_"
+    };
+    GlobalMetadataDBConfig config;
+    auto const vm = parse_args(argv, config);
+
+    REQUIRE(config.get_metadata_db_type() == GlobalMetadataDBConfig::MetadataDBType::MySQL);
+    REQUIRE(config.get_metadata_db_host() == "test-host");
+    REQUIRE(config.get_metadata_db_port() == 8888);
+    REQUIRE(config.get_metadata_db_name() == "test-db");
+    REQUIRE(config.get_metadata_table_prefix() == "test_prefix_");
+}
+
+TEST_CASE("Test MySQL arguments and credential validation", "[GlobalMetadataDBConfig]") {
+    constexpr std::array argv{
+            "test",
+            "--db-type",
+            "mysql",
+            "--db-host",
+            "test-host",
+            "--db-port",
+            "8888",
+            "--db-name",
+            "test-db",
+            "--db-table-prefix",
+            "test_prefix_"
+    };
+    GlobalMetadataDBConfig config;
+    auto const vm = parse_args(argv, config);
+
+    SECTION("With valid credentials") {
+        set_env_var("CLP_DB_USER", "test-user");
+        set_env_var("CLP_DB_PASS", "test-pass");
+
+        config.read_credentials_from_env_if_needed();
+        REQUIRE_NOTHROW(config.validate());
+        REQUIRE(config.get_metadata_db_username() == "test-user");
+        REQUIRE(config.get_metadata_db_password() == "test-pass");
+
+        unset_env_var("CLP_DB_USER");
+        unset_env_var("CLP_DB_PASS");
+    }
+
+    SECTION("With missing credentials") {
+        SECTION("Neither username nor password") {
+            config.read_credentials_from_env_if_needed();
+            REQUIRE_THROWS_AS(config.validate(), std::invalid_argument);
+        }
+        SECTION("Username set but password missing") {
+            set_env_var("CLP_DB_USER", "test-user");
+            config.read_credentials_from_env_if_needed();
+            REQUIRE_THROWS_AS(config.validate(), std::invalid_argument);
+            unset_env_var("CLP_DB_USER");
+        }
+        SECTION("Password set but username missing") {
+            set_env_var("CLP_DB_PASS", "test-pass");
+            config.read_credentials_from_env_if_needed();
+            REQUIRE_THROWS_AS(config.validate(), std::invalid_argument);
+            unset_env_var("CLP_DB_PASS");
+        }
+    }
+}
+
+TEST_CASE("Test SQLite arguments", "[GlobalMetadataDBConfig]") {
+    constexpr std::array argv{"test", "--db-type", "sqlite"};
+    GlobalMetadataDBConfig config;
+    auto const vm = parse_args(argv, config);
+
+    set_env_var("CLP_DB_USER", "test-user");
+    set_env_var("CLP_DB_PASS", "test-pass");
+
+    config.read_credentials_from_env_if_needed();
+    REQUIRE(config.get_metadata_db_username().empty());
+    REQUIRE(config.get_metadata_db_password().empty());
+    REQUIRE_NOTHROW(config.validate());
+
+    unset_env_var("CLP_DB_USER");
+    unset_env_var("CLP_DB_PASS");
+}
```

Comment:
can we add another section to assert that when db-type is sqlite, mysql argument cause the argparser to throw?